frontends/election-manager/src/components/full_test_deck_tally_report_button.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useContext, useEffect } from 'react';
+import { assert, tallyVotesByContest } from '@votingworks/utils';
+import { LogEventId } from '@votingworks/logging';
+import { Tally, VotingMethod } from '@votingworks/types';
+
+import { AppContext } from '../contexts/app_context';
+import { PrintableArea } from './printable_area';
+import { PrintButton } from './print_button';
+import { TestDeckTallyReport } from './test_deck_tally_report';
+import { generateTestDeckBallots } from '../utils/election';
+
+export function FullTestDeckTallyReportButton(): JSX.Element {
+  const { currentUserSession, electionDefinition, logger } =
+    useContext(AppContext);
+
+  assert(currentUserSession);
+  const currentUserType = currentUserSession.type;
+
+  assert(electionDefinition);
+  const { election } = electionDefinition;
+
+  const ballots = generateTestDeckBallots({ election });
+  const votes = ballots.map((b) => b.votes);
+  const testDeckTally: Tally = {
+    numberOfBallotsCounted: ballots.length * 4,
+    castVoteRecords: new Set(),
+    contestTallies: tallyVotesByContest({
+      election,
+      votes: [...votes, ...votes, ...votes, ...votes],
+    }),
+    ballotCountsByVotingMethod: { [VotingMethod.Unknown]: ballots.length },
+  };
+
+  function disablePrinting() {
+    document
+      .getElementById('full-test-deck-tally-report')
+      ?.classList.add('no-print');
```

Comment:
This is an effect, and so should be happening inside a `useEffect`. That said, this is more appropriately modeled using state.